p2p/net/swarm/limiter_test.go

```diff
@@ -0,0 +1,315 @@
+package swarm
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	peer "github.com/ipfs/go-libp2p-peer"
+	ma "github.com/jbenet/go-multiaddr"
+	mafmt "github.com/whyrusleeping/mafmt"
+	context "golang.org/x/net/context"
+
+	conn "github.com/ipfs/go-libp2p/p2p/net/conn"
+)
+
+func mustAddr(t *testing.T, s string) ma.Multiaddr {
+	a, err := ma.NewMultiaddr(s)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return a
+}
+
+func addrWithPort(t *testing.T, p int) ma.Multiaddr {
```

Comment:
It's not super important, but you could have the funcs `generateGoodAddr`, `generateBadAddr`, and `isGoodAddr` that use some magic port number underneath but don't expose it.